assets/php/class-wgpb-block-library.php

```diff
@@ -54,7 +54,28 @@ public static function init() {
 		self::register_blocks();
 		self::register_assets();
 		add_filter( 'block_categories', array( 'WGPB_Block_Library', 'add_block_category' ) );
+		add_filter( 'woocommerce_enqueue_styles', array( 'WGPB_Block_Library', 'enqueue_styles' ), 20 );
 		add_action( 'admin_print_footer_scripts', array( 'WGPB_Block_Library', 'print_script_settings' ), 1 );
+		add_action( 'body_class', array( 'WGPB_Block_Library', 'add_theme_body_class' ), 1 );
+	}
+
+	/**
+	 * Add block styles to WooCommmerce.
+	 *
+	 * @internal This is here instead of register_block_type so styles can be unhooked, and so they are loaded after Woo core styles.
+	 *
+	 * @param array $styles Array of Woo core stylesheets.
+	 * @return array
+	 */
+	public static function enqueue_styles( $styles ) {
+		$styles['woocommerce-blocks'] = array(
```

Comment:
What if we leave the style enqueued in `register_block_type`, but dequeue it here right before re-registering it this way? Would that work?

Reply:
Would it hurt to move all styles into the main editor.scss and style.scss? Since they are all built and merged into one anyway, is there much advantage is allowing local styles?

Reply:
it's pretty standard for react projects, both calypso & gutenberg use this pattern of styles living next to the components they style.